.github/workflows/test_iree.yml

```diff
@@ -215,4 +215,10 @@ jobs:
           steps.unet.outcome == 'success' && steps.vae.outcome == 'success'
         run: |
           source ${VENV_DIR}/bin/activate
-          bash iree_tests/benchmarks/benchmark_sdxl_rocm.sh
+          pytest iree_tests/benchmarks/benchmark_sdxl_rocm.py --goldentime 5565
+    
+      - name: Upload SDXL ROCm benchmark artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: sdxl-benchmark-output
+          path: iree_tests/benchmarks/benchmark_out_rocm.txt
```

Comment:
> Also, I added an artifact that gets uploaded, so that the user can look through every iteration and the other data provided as well (median, std dev, cv, user counters, etc.) for the sdxl benchmark: https://github.com/nod-ai/SHARK-TestSuite/actions/runs/9381843307/artifacts/1570376307

We can improve the utility here. This is uploading a human readable format to a file that still needs a few steps to view:
1. go to results page
2. download artifact
3. unzip
4. open text file

## Uploading artifacts

If uploading output, you could choose a different output format so automated systems can more easily process it: https://github.com/google/benchmark/blob/main/docs/user_guide.md#output-formats

Eventually I'd want benchmark results to be uploaded to a database. The in-tree benchmarking that IREE has uploads to https://perf.iree.dev/ but the underlying database (through https://github.com/google/dana) is not structured in a way that it can be easily queried.

## Posting comments

Another option is posting comments to the PR (like with IREE's [`post_benchmark_comment.yaml`](https://github.com/iree-org/iree/blob/main/.github/workflows/post_benchmark_comment.yaml)). That would skip all 4 steps above but I'd prefer for that to only happen if the PR author opted in or the results were particularly noteworthy. A PR touching Python bindings or updating documentation shouldn't get a comment about regressed benchmarks, for example.

## Job summaries

Some middle ground is a job summary:
* https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary
* https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/

you'd generate some markdown that would appear on https://github.com/nod-ai/SHARK-TestSuite/actions/runs/9381843307/ , like https://github.com/iree-org/iree/actions/runs/9387594823#summary-25853165697 . That skips steps 2-4 above but still needs step 1.

## My recommendation

For now I'd suggest posting a job summary with human readable text (up to half of a screen or so? shouldn't be too noisy and should plan for more benchmark data being shown in the same location, such as from multiple machines or multiple models). You could keep the uploaded file either as the "console" format (default) or "json".

Reply:
Did not know job summary was a thing. Thanks, seems like this is the best/cleanest fit for the benchmarking setup right now.